### PR TITLE
docs: Artifact hub improvements

### PR DESCRIPTION
**Pull request by james-milligan**
## This PR

- improves the artefact hub documentation 
- nests the operator chart, matching the pattern commonly used by other entries in the artefact hub

### Related Issues

https://github.com/open-feature/open-feature-operator/issues/231

### Follow-up Tasks

Once the artefact hub account is set up for OpenFeature, the existing entry needs to be removed and a new one set up to point to this repo. The icon will also be setup within the account config.

if this pr https://github.com/open-feature/open-feature-operator/pull/278 merges first the helm installation instructions also need to be updated

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ spec:
         name: manager
         env:
         - name: FLAGD_VERSION
-          value: INPUT_FLAGD_VERSION
+          value: v0.2.7
```
Comment (skyerus) [inline config/manager/manager.yaml:39]: Shouldn't be checked in

Comment (james-milligan) [reply config/manager/manager.yaml]: resolved

**Comment by AlexsJones** [comment]
Have you tested this works @james-milligan ? From what I understand using the FQDN of `open-feature-operator` with helm will result in a name that's too long for K8s admission.

**Comment by james-milligan** [comment]
> Have you tested this works @james-milligan ? From what I understand using the FQDN of open-feature-operator with helm will result in a name that's too long for K8s admission.

Ive set it up to test on my forked gh-pages here and am able to install the operator https://artifacthub.io/packages/helm/open-feature-operator-dev/open-feature-operator

File: chart/open-feature-operator/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  / _ \   _ __   ___   _ _   | __|  ___   __ _  | |_   _  _   _ _   ___ 
+ | (_) | | '_ \ / -_) | ' \  | _|  / -_) / _` | |  _| | || | | '_| / -_)
+  \___/  | .__/ \___| |_||_| |_|   \___| \__,_|  \__|  \_,_| |_|   \___|
+         |_|                                                            
+
+Thank you for installing {{ .Chart.Name }}.
+
+Your release version is {{ .Chart.AppVersion }}.
+
+Documentation and configuration options can be found here: https://github.com/open-feature/open-feature-operator
```
Comment (toddbaert) [inline chart/open-feature-operator/templates/NOTES.txt:11]: :star_struck:

**Review by toddbaert — APPROVED** [review]
lgtm

File: config/manager/manager.yaml (hunk 1 above)
Comment (beeme1mr) [reply config/manager/manager.yaml]: Any idea how this happens? Can this accidental change be avoided in the future with an update the tooling?

Comment (james-milligan) [reply config/manager/manager.yaml]: I missed this while building the chart for testing, ill look into it to see if we can avoid this happening again

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**